clients/include/testing_csr2bsr.hpp

```diff
@@ -0,0 +1,703 @@
+    {
+        filename = argus.filename;
+    }
+
+    std::unique_ptr<handle_struct> unique_ptr_handle(new handle_struct);
+    hipsparseHandle_t              handle = unique_ptr_handle->handle;
+    std::unique_ptr<descr_struct> unique_ptr_csr_descr(new descr_struct);
+    hipsparseMatDescr_t           csr_descr = unique_ptr_csr_descr->descr;
+    std::unique_ptr<descr_struct> unique_ptr_bsr_descr(new descr_struct);
+    hipsparseMatDescr_t           bsr_descr = unique_ptr_bsr_descr->descr;
+
+    hipsparseSetMatIndexBase(csr_descr, csr_idx_base);
+    hipsparseSetMatIndexBase(bsr_descr, bsr_idx_base);
+
+    // Argument sanity check before allocating invalid memory
+    if(m <= 0 || n <= 0 || block_dim <= 0)
+    {
+#ifdef __HIP_PLATFORM_NVCC__
+        // Do not test args in cusparse
+        return HIPSPARSE_STATUS_SUCCESS;
+#endif
+        auto dcsr_row_ptr_managed
+            = hipsparse_unique_ptr{device_malloc(sizeof(int) * safe_size), device_free};
+        auto dcsr_col_ind_managed
+            = hipsparse_unique_ptr{device_malloc(sizeof(int) * safe_size), device_free};
+        auto dcsr_val_managed
+            = hipsparse_unique_ptr{device_malloc(sizeof(T) * safe_size), device_free};
+        auto dbsr_row_ptr_managed
+            = hipsparse_unique_ptr{device_malloc(sizeof(int) * safe_size), device_free};
+        auto dbsr_col_ind_managed
+            = hipsparse_unique_ptr{device_malloc(sizeof(int) * safe_size), device_free};
+        auto dbsr_val_managed
+            = hipsparse_unique_ptr{device_malloc(sizeof(T) * safe_size), device_free};
+
+        int* dcsr_row_ptr = (int*)dcsr_row_ptr_managed.get();
+        int* dcsr_col_ind = (int*)dcsr_col_ind_managed.get();
+        T*   dcsr_val     = (T*)dcsr_val_managed.get();
+        int* dbsr_row_ptr = (int*)dbsr_row_ptr_managed.get();
+        int* dbsr_col_ind = (int*)dbsr_col_ind_managed.get();
+        T*   dbsr_val     = (T*)dbsr_val_managed.get();
+
+        if(!dcsr_row_ptr || !dcsr_col_ind || !dcsr_val || !dbsr_row_ptr || !dbsr_col_ind || !dbsr_val)
+        {
+            verify_hipsparse_status_success(HIPSPARSE_STATUS_ALLOC_FAILED,
+                                            "!dcsr_row_ptr || !dcsr_col_ind || !dcsr_val || "
+                                            "!dbsr_row_ptr || !dbsr_col_ind || !dbsr_val || ");
+            return HIPSPARSE_STATUS_ALLOC_FAILED;
+        }
+
+        int bsr_nnzb;
+
+        status = hipsparseXcsr2bsr_nnz(handle,
+                                       dir,
+                                       m,
+                                       n,
+                                       csr_descr,
+                                       dcsr_row_ptr,
+                                       dcsr_col_ind,
+                                       block_dim,
+                                       bsr_descr,
+                                       dbsr_row_ptr,
+                                       &bsr_nnzb);
+
```

Comment:
Note to self. Add verify hipsparse status after csr2bsr_nnz call as well:

if(m < 0 || n < 0 || block_dim < 0)
        {
            verify_hipsparse_status_invalid_size(status, "Error: m < 0 || n < 0 || block_dim < 0");
        }
        else
        {
            verify_hipsparse_status_success(status, "m >= 0 && n >= 0 && block_dim >= 0");
        }